src/common/buffer.cc

```diff
@@ -73,6 +74,9 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
       buffer_history_alloc_num.inc();
     }
   }
+    int close(int fd) {
```

Comment:
wrong indent. might want to add an empty line before this function. also, i'd suggest just use `TEMP_FAILURE_RETRY(::close(fd))` directly instead introducing a wrapper for it.